name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/earnings/EarningsListWidget.java

```diff
@@ -275,14 +329,57 @@ public void update(Model model)
         ((DashboardView) this.view).updateScrolledCompositeMinSize();
     }
 
+    private void createCollapsedArray()
+    {
+        ExpansionSetting expansion = get(ExpansionSettingConfig.class).getValue();
+        collapsed = new boolean[12];
+
+        if (expansion.equals(ExpansionSetting.EXPAND_ALL))
+        {
+            return; //
+        }
+        Arrays.fill(collapsed, true);
+        if (expansion.equals(ExpansionSetting.COLLAPSE_ALL))
+        {
+            return; //
+        }
+        LocalDate now = LocalDate.now();
+
+        collapsed[now.getMonthValue() - 1] = false;
+    }
+
     protected Composite createMonthHeader(Composite parent, MonthData month)
     {
         Composite composite = new Composite(parent, SWT.NONE);
         composite.setLayout(new FormLayout());
 
         Label label = new Label(composite, SWT.NONE);
         label.setData(UIConstants.CSS.CLASS_NAME, UIConstants.CSS.HEADING2);
-        label.setText(monthLabels[month.monthOfYear - 1]);
+        int mVal = month.monthOfYear - 1;
+        label.setText(monthLabels[mVal]);
+        label.setCursor(new HyperlinkSettings(parent.getDisplay()).getHyperlinkCursor());
+        label.addMouseListener(new MouseListener()
+        {
+
+            @Override
+            public void mouseUp(MouseEvent e)
+            {
+                collapsed[mVal] = !collapsed[mVal];
+                update();
+            }
+
+            @Override
+            public void mouseDown(MouseEvent e)
+            {
+                // empty, we're waiting for mouse-up
+            }
+
+            @Override
+            public void mouseDoubleClick(MouseEvent e)
+            {
+                // maybe another time with another function behind it
+            }
+        });
```

Comment:
BTW, you can use also the adapters if we anyway implement only one method

```java
        label.addMouseListener(MouseListener.mouseUpAdapter(e -> {
            collapsed[mVal] = !collapsed[mVal];
            update();
        }));
```

Reply:
I knew that there was something like this... ;-)